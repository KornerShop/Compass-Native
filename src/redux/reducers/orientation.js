import initState from '../initialState';
import { UPDATE_ORIENTATION } from '../actions/types';

export default (state = initState.orientation, { type, payload }) => {
  switch (type) {
    case UPDATE_ORIENTATION:
      return payload;
    default:
      return state;
  }
};
