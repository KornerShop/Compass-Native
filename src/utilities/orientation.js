export default dimensions => {
	let view
	const {height} = dimensions
	const {width} = dimensions
	if (!dimensions.height || !dimensions.width) {
		throw new Error(
			'landscape method requires an object with properties height and width'
		)
	}
	if (dimensions.height < dimensions.width) {
		view = 'landscape'
	} else {
		view = 'portrait'
	}
	return {
		view,
		height,
		width,
	}
}
