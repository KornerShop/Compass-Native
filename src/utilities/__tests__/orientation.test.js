import orientation from '../orientation';

test('orientation', () => {
  const position = orientation({
    width: 411.42857142857144,
    height: 683.4285714285714,
    scale: 2.625,
    fontScale: 1,
  });
  expect(position).toMatchSnapshot();
});
