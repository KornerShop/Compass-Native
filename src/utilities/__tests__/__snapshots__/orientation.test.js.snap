// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`orientation 1`] = `
Object {
  "height": 683.4285714285714,
  "view": "portrait",
  "width": 411.42857142857144,
}
`;
